synthesizer/program/src/lib.rs

```diff
@@ -912,6 +912,52 @@ impl<N: Network> ProgramCore<N> {
         // Return `false` since no V9 syntax was found.
         false
     }
+
+    /// Returns `true` if the program contains an array type with a size that exceeds the given maximum.
+    pub fn exceeds_max_array_size(&self, max_array_size: u32) -> bool {
+        self.mappings.values().any(|mapping| mapping.exceeds_max_array_size(max_array_size))
```

Comment:
Can mapping keys also be arrays? `MapKey` contains a generic `PlaintextType` so maybe we should just check.